### PR TITLE
Propagate FS errors when creating a new project

### PR DESCRIPTION
**Pull request by radeusgd**
### Pull Request Description

- Part of #10662 

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.

### CODE REVIEW

File: lib/scala/pkg/src/main/scala/org/enso/pkg/Package.scala [hunk 1]
```diff
@@ -265,7 +263,7 @@ class PackageManager[F](implicit val fileSystem: FileSystem[F]) {
     template: Template
   ): Package[F] = {
     val pkg = new Package(root, config, fileSystem)
-    pkg.save()
+    pkg.save().get
```
Comment (radeusgd) [inline lib/scala/pkg/src/main/scala/org/enso/pkg/Package.scala]: Discarding the result of `Try[...]`... - not the best idea

File: lib/scala/pkg/src/main/scala/org/enso/pkg/Package.scala [hunk 2]
```diff
@@ -9,7 +9,18 @@ import java.net.URI
 import scala.jdk.CollectionConverters._
 import scala.util.{Failure, Try, Using}
 
-object CouldNotCreateDirectory extends Exception
+case class CouldNotCreateDirectory(cause: Throwable) extends RuntimeException {
+  override def getMessage: String =
+    s"Could not create directory: ${cause.getMessage}. Perhaps there is a permission issue."
```
Comment (radeusgd) [inline lib/scala/pkg/src/main/scala/org/enso/pkg/Package.scala:15]: I added the comment, as for me this was caused by https://learn.microsoft.com/en-us/defender-endpoint/controlled-folders#what-is-controlled-folder-access

File: engine/runner/src/main/java/org/enso/runner/Main.java [hunk 3]
```diff
@@ -1383,7 +1383,7 @@ private void launch(Options options, CommandLine line, Level logLevel, boolean l
                 return BoxedUnit.UNIT;
               });
         } catch (IOException ex) {
-          System.err.println(ex.getMessage());
+          ex.printStackTrace();
```
Comment (radeusgd) [inline engine/runner/src/main/java/org/enso/runner/Main.java]: `getMessage` is sometimes unhelpful for Java exceptions, e.g. `NoSuchFileException::getMessage` returns just the path - but no comment - so from `getMessage` you have no idea what happened, what kind of exception this was - you just see a file path and wonder where did this come from...

The stack trace should be more helpful I think.

**Review by GregoryTravis — APPROVED** [review]

File: engine/runner/src/main/java/org/enso/runner/Main.java (hunk 3 above)
Comment (hubertp) [reply engine/runner/src/main/java/org/enso/runner/Main.java]: If so, the full stacktrace should be printed only when verbose mode is on. We don't want to show this to end users.

File: lib/scala/pkg/src/main/scala/org/enso/pkg/Package.scala (hunk 1 above)
Comment (hubertp) [reply lib/scala/pkg/src/main/scala/org/enso/pkg/Package.scala]: What's the point of using `Try` there, then? `.get` will crash with an exception on failure and not really report the failure either.

**Review by hubertp — CHANGES_REQUESTED** [review]

File: engine/runner/src/main/java/org/enso/runner/Main.java (hunk 3 above)
Comment (radeusgd) [reply engine/runner/src/main/java/org/enso/runner/Main.java]: What is the alternative? As mentioned, `getMessage` is utterly useless. I guess I could try `toString`.

File: engine/runner/src/main/java/org/enso/runner/Main.java [hunk 4]
```diff
@@ -1385,7 +1385,8 @@ private void launch(CommandLine line, Level logLevel, boolean logMasking) {
                 return BoxedUnit.UNIT;
               });
         } catch (IOException ex) {
-          System.err.println(ex.getMessage());
+          logger.error("Error during execution", ex);
```
Comment (hubertp) [inline engine/runner/src/main/java/org/enso/runner/Main.java]: You could check `if (Logger.isEnabledForLevel(Level.DEBUG)) { ...`

**Review by JaroslavTulach — APPROVED** [review]

**Review by hubertp — APPROVED** [review]

**Pull request merged**